src/async-wrap.cc

```diff
@@ -121,18 +122,35 @@ static void SetupHooks(const FunctionCallbackInfo<Value>& args) {
 
   if (env->async_hooks()->callbacks_enabled())
```

Comment:
Right, question is why is it the responsibility of the `setupHooks()` caller to `enable()/disable()`?  i.e., why can't this be done implicitly by `setupHooks()`?  

Same question follows when the handlers are invoked.  We know that if an `init()` callback triggers an async operation, it risks infinite recursion, so why not implicitly disable async callbacks before invoking `init()`?